x-pack/plugins/transform/common/types/index_pattern.ts

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { IndexPattern } from '../../../../../src/plugins/data/common';
+
+import { isPopulatedObject } from '../utils/object_utils';
+
+// Custom minimal type guard for IndexPattern to check against the attributes used in transforms code.
+export function isIndexPattern(arg: any): arg is IndexPattern {
```

Comment:
Unfortunately that doesn't work because `src/plugins/data` uses `export type` only, and we cannot do a deep import from `src/plugins/data/common/index_patterns/index_patterns/index_pattern`.